website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,154 @@
+retrieve_objects = ['test1.xml', 'test2.xml']
+
+s3 = boto3.client('s3')
+for retrieve_object in retrieve_objects:
+    try:
+        response = s3.restore_object(
+                Bucket='orca-rhassan-sandbox-glacier', # bucket that contains the objects currently in glacier flexible retrieval
+                Key = retrieve_object,
+                RestoreRequest={
+                    'Days': 30,
+                    'GlacierJobParameters': {
+                        'Tier': 'Bulk',
+                    },
+                },
+            )
+        print(response)
+    except Exception as ex:
+        print(f"{ex} for {retrieve_object}")
+
+```
+Output:
+```
+{'ResponseMetadata': {'RequestId': '61ZWC27NHSFNMYJT', 'HostId': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'x-amz-request-id': '61ZWC27NHSFNMYJT', 'date': 'Sun, 24 Jul 2022 18:01:33 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+{'ResponseMetadata': {'RequestId': 'PNN4XXA5SYTBBRKG', 'HostId': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'x-amz-request-id': 'PNN4XXA5SYTBBRKG', 'date': 'Sun, 24 Jul 2022 18:01:34 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+```
+
+Also see existing ORCA [restore_object function](https://github.com/nasa/cumulus-orca/blob/develop/tasks/request_files/request_files.py#L548)
+Make sure the S3 bucket policy has permission for `s3:PutObject*` action otherwise user will see access denied error. See this [policy](https://github.com/nasa/cumulus-orca/blob/develop/website/docs/developer/deployment-guide/creating-orca-glacier-bucket.md#archive-bucket) for additional details.
+
+Once all files are restored from glacier, there should be some mechanism that will notify users that the retrieval process is complete. One approach could be to enable event notifications in bucket that will trigger either an SQS, SNS or lambda. This will require additional research since AWS notifies for each `s3:ObjectRestore:Completed` API call and not for a specific number of objects.
+
+- Once all objects are retrieved in standard S3 storage, the bucket's lifecycle policy can be configured with `Deep Archive` storage type if the whole bucket objects have to be moved to deep archive.
+
+```teraform
+resource "aws_s3_bucket_lifecycle_configuration" "bucket-config" {
+  bucket = aws_s3_bucket.bucket.bucket
+
+  rule {
+    id = "transition to deep archive"
+    status = "Enabled"
+
+    transition {
+      days          = 0
+      storage_class = "DEEP_ARCHIVE"
+    }
+  }
+
+```
+:::warning
+This will not work in case a user only wants to migrate some specific objects to deep archive.
```

Comment:
It also causes problems if the user doesn't want all of their future files to be placed in DEEP_ARCHIVE.

Reply:
Might just need a section on removing the rule.

Reply:
I think the requirement/business logic was to move to DEEP ARCHIVE.

Reply:
Do we assume that all future files should be placed in DEEP_ARCHIVE?

If so, we need to modify ORCA to check for this lifecycle rule so it knows that it cannot store to anything other than DEEP_ARCHIVE (not recommended).

If not, then this lifecycle rule needs to be removed after the migration is complete.

Reply:
added note on using a different storage class other than deep archive

Reply:
The issue is that leaving the lifecycle_configuration in place will automatically move any NEW files into DEEP_ARCHIVE, regardless of instructions on ingest.

Reply:
added note to disable the lifecycle rule after migration